docs/src/filters/matches.md

```diff
@@ -0,0 +1,45 @@
+# Matches
+
+The `Matches` filter's job is to provide a mechanism to change behaviour based
+on dynamic metadata. This filter behaves similarly to the `match` expression
+in Rust or `switch` statements in other languages.
+
+#### Filter name
+```text
+quilkin.extensions.filters.matches.v1alpha1.Matches
+```
+
+### Configuration Examples
+```rust
+# let yaml = "
+version: v1alpha1
+static:
+  endpoints:
+    - address: 127.0.0.1:26000
+    - address: 127.0.0.1:26001
+  filters:
+    - name: quilkin.extensions.filters.capture_bytes.v1alpha1.CaptureBytes
+      config:
+          strategy: PREFIX
+          metadataKey: myapp.com/token
+          size: 3
+          remove: false
+    - name: quilkin.extensions.filters.matches.v1alpha1.Matches
+      config:
+          on_read:
+            metadataKey: myapp.com/token
+            branches:
+                - value: abc
```

Comment:
From digging into the code, `value` can't be a base64 byte array, correct? It has be one of the `Value` types such as String or Int? (bool?). This makes sense, since the dynamic metadata could come from anywhere, and as such, should be a `Value`.

(Just making sure my understanding is correct).

If so, we might want to make that extra clear in the documentation.

Reply:
Yes, it's not base64, it's a string.

Reply:
_nods_ that makes total sense -- I think we should be clear in that in the docs with a comment somewhere, as I could see people assuming it was a byte array and attempting to fill it as such.

Reply:
Well I mean it should be documented but that's also valid to do that. That's why the `PartialEq` of `Value` compares the value between strings and bytes, so that you can set a string here and use it to match the bytes, since a lot of marker values in my experience are at least string friendly if not human-readable.

Reply:
100% agree.